src/Dapr.PluggableComponents/Components/SecretStores/SecretStoreBulkGetResponse.cs

```diff
@@ -0,0 +1,43 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2023 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using Dapr.PluggableComponents.Utilities;
+using Dapr.Proto.Components.V1;
+using Google.Protobuf;
+
+namespace Dapr.PluggableComponents.Components.SecretStores;
+
+public sealed record SecretStoreBulkGetResponse
+{
+    public IReadOnlyDictionary<string, SecretStoreResponse> Data { get; init; } = new Dictionary<string, SecretStoreResponse>();
+
+    internal static BulkGetSecretResponse ToBulkGetResponse(SecretStoreBulkGetResponse response)
+    {
+        var grpcResponse = new BulkGetSecretResponse();
+        if (response != null)
+        {
+            foreach (var item in response.Data)
+            {
+                var secretResp = new SecretResponse();
+                foreach (var sec in item.Value.Data)
+                {
+                    secretResp.Secrets.Add(sec.Key, sec.Value);
+                    grpcResponse.Data.Add(sec.Key, secretResp);
```

Comment:
This should only happen once, right, outside the `foreach` and for the `item.Key` key?